expr/types.go

```diff
@@ -0,0 +1,91 @@
+// argument types. to let functions describe their inputs and outputs
+package expr
+
+type arg interface {
```

Comment:
what's the specific reason? do you anticipate people will want to write functions in different packages (as opposed to a fork of this package?)